### PR TITLE
Fix #3101. Add dropdownAutoWidth option.

### LINKED ISSUE #3101: select2 4.0 - dropdownAutoWidth not working
option dropdownAutoWidth is not working in 4.0. It's important in my case...

### PR DESCRIPTION
**Pull request by menthol**
Add the dropdownAutoWidth options. This option force the minimal width to be the same than the select width.

### CODE REVIEW

**Comment by kevin-brown** [comment]
This is going to need documentation as well.

I wouldn't mind seeing a jsbin that demonstrates how this works (in this pull request) so I can do some quick testing on how it compares to the old "measure the width" method.

**Comment by j-a-p-dev** [comment]
I'm sorry I didn't known how to create a jsbin with a pull request.

I make this to show how I use it. I just add a option for max dropdown width

http://jsbin.com/licimafagu/2/edit

**Comment by menthol** [comment]
I'm sorry I didn't known how to create a jsbin with a pull request.

I make this to show how I use it. I just add a option for max dropdown width

http://jsbin.com/licimafagu/2/edit

File: src/js/select2/dropdown/attachBody.js [hunk 1]
```diff
@@ -177,11 +179,17 @@ define([
...
-
-    this.$dropdown.css({
+    var css = {
       width: this.$container.outerWidth(false) + 'px'
-    });
+    };
+
+    if (this.dropdownAutoWidth)
+    {
+      this.$dropdownContainer.width(this.dropdownMaxWidth);
```
Comment (kevin-brown) [inline src/js/select2/dropdown/attachBody.js]: Shouldn't this just be done with `css.maxWidth`?

File: src/js/select2/defaults.js [hunk 2]
```diff
@@ -318,6 +318,8 @@ define([
       amdBase: 'select2/',
       amdLanguageBase: 'select2/i18n/',
       closeOnSelect: true,
+      dropdownAutoWidth: false,
+      dropdownMaxWidth: '100%',
```
Comment (kevin-brown) [inline src/js/select2/defaults.js]: If you can get away with using `css.maxWidth` below, then you should be able to just not include this in the defaults.

**Comment by kevin-brown** [comment]
I've left a few inline comments. This still needs documentation (or an example, might be easier) before I can accept this pull request.

> http://jsbin.com/licimafagu/2/edit

Awesome, looks good.

**Comment by menthol** [comment]
The purpose of this pull request is to be able to have a dropdown with a width that isn't the width of the select. maxWidth is the max width of the select so it will be blocked at a size tinier than expected one.

**Comment by btaens** [comment]
Any development on this front?
A pretty important feature.

**Comment by kevin-brown** [comment]
This is going to need rebasing, but I may try to manually merge it later tonight.

**Comment by KODerFunk** [comment]
:thumbsup: Still need!

**Pull request closed without merging**